### PR TITLE
[DEV-3791] Add parent_award to contract award response

### PR DESCRIPTION
**Pull request by sethstoudenmier**
**Description:**
Currently, the IDV award response for `/api/v2/awards/{award_id}` returns an object in the response for `parent_award`. This PR adds this object to the Contract award response. In the process, the awards endpoint response for Contract and IDV awards was cleaned up a little (reflected by the API Contract). The changes are:

* Moved `parent_genereated_unique_award_id` to be `generated_unique_award_id` inside of the `parent_award` object
* Removed `parent_award_piid` since you can find `piid` inside of the `parent_award` object
* Added the `parent_award` object to the response for Contract awards

**Technical details:**
Added the `parent_award` object to the awards response for Contract awards. 

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [X] Necessary PR reviewers:
    - [X] Backend
    - [X] Frontend <OPTIONAL>
4. [X] Matview impact assessment completed (N/A)
5. [X] Frontend impact assessment completed
6. [X] Data validation completed
7. [X] Appropriate Operations ticket(s) created (N/A)
8. [X] Jira Ticket [DEV-3791](https://federal-spending-transparency.atlassian.net/browse/DEV-3791):
    - [X] Link to this Pull-Request
    - [X] Performance evaluation of affected (API | Script | Download)
    - [X] Before / After data comparison

Changes to the Contract response:
```{
    "subaward_count": 0,
    "generated_unique_award_id": "CONT_AWD_00004200011C401811Y018401819D010_1448_401819D010_1448",
    "description": null,
    "type": "C",
    "base_and_all_options": 0.0,
    "base_exercised_options": 0.0,
    "total_obligation": 1000.0,
    "type_description": "DO",
    "id": 12500,
    "category": "contract",
    "piid": "00004200011C401811Y018401819D010",
    "date_signed": "2000-11-15",
    "total_subaward_amount": null,
    "parent_award": {
        "agency_id": "1448",
        "agency_name": "U.S. Fish and Wildlife Service",
        "award_id": 68839000,
        "generated_unique_award_id": "CONT_IDV_401819D010_1448",
        "idv_type_description": "IDC",
        "multiple_or_single_aw_desc": "SINGLE AWARD",
        "piid": "401819D010",
        "type_of_idc_description": "NAN"
    },
    ...
}
```

Changes to the IDV response:
```
{
    "subaward_count": 0,
    "generated_unique_award_id": "CONT_IDV_GS07FDA450_4732",
    "description": "OTHER THAN SCHEDULE",
    "type": "IDV_E",
    "base_and_all_options": 149999.0,
    "base_exercised_options": null,
    "total_obligation": 0.0,
    "type_description": "BPA",
    "id": 69254500,
    "category": "idv",
    "piid": "GS07FDA450",
    "date_signed": "2016-09-22",
    "total_subaward_amount": null,
    "parent_award": {
        "agency_id": "4732",
        "agency_name": "Federal Acquisition Service",
        "award_id": 68942500,
        "generated_unique_award_id": "CONT_IDV_GS07F0301X_4732",
        "idv_type_description": "FSS",
        "multiple_or_single_aw_desc": "MULTIPLE AWARD",
        "piid": "GS07F0301X",
        "type_of_idc_description": null
    },
    ...
}
```

Contract response before the changes:
```
{
    "subaward_count": 0,
    "base_and_all_options": 0.0,
    "id": 12500,
    "description": null,
    "date_signed": "2000-11-15",
    "category": "contract",
    "total_obligation": 1000.0,
    "piid": "00004200011C401811Y018401819D010",
    "base_exercised_options": 0.0,
    "parent_award_piid": "401819D010",
    "total_subaward_amount": null,
    "type": "C",
    "generated_unique_award_id": "CONT_AWD_00004200011C401811Y018401819D010_1448_401819D010_1448",
    "type_description": "DO",
    "latest_transaction_contract_data": {
        "consolidated_contract": "N",
        "cost_or_pricing_data_description": null,
        "fair_opportunity_limited_description": "NAN",
        "interagency_contracting_authority_description": "NOT APPLICABLE",
        "dod_acquisition_program": null,
        "cost_or_pricing_data": null,
        "product_or_service_code": "R402",
        "dod_claimant_program": null,
        "materials_supplies_description": "NOT APPLICABLE",
        "purchase_card_as_payment_method": "N",
        "foreign_funding_description": "NOT APPLICABLE",
        "subcontracting_plan_description": "PLAN NOT REQUIRED",
        "small_business_competitive": false,
        "sea_transportation_description": null,
        "extent_competed_description": "FULL AND OPEN COMPETITION",
        "multiple_or_single_award_description": null,
        "solicitation_procedures": null,
        "subcontracting_plan": "B",
        "foreign_funding": null,
        "price_evaluation_adjustment": "0",
        "materials_supplies": "X",
        "type_set_aside_description": "SMALL BUSINESS SET ASIDE - TOTAL",
        "labor_standards_description": "YES",
        "commercial_item_acquisition_description": null,
        "number_of_offers_received": "1",
        "interagency_contracting_authority": "X",
        "referenced_idv_agency_desc": "U.S. FISH AND WILDLIFE SERVICE",
        "multi_year_contract_description": "NO",
        "clinger_cohen_act_planning_description": null,
        "type_of_contract_pricing": "J",
        "evaluated_preference": null,
        "solicitation_identifier": null,
        "major_program": null,
        "fed_biz_opps_description": "YES",
        "labor_standards": "Y",
        "product_or_service_description": "REAL ESTATE BROKERAGE SERVICES",
        "commercial_item_acquisition": "D",
        "naics": "524127",
        "type_of_contract_pricing_description": "FIRM FIXED PRICE",
        "multi_year_contract": "N",
        "type_of_idc_description": "NAN",
        "evaluated_preference_description": null,
        "domestic_or_foreign_entity_description": null,
        "dod_acquisition_program_description": null,
        "commercial_item_test_program_description": "NO",
        "other_than_full_and_open_description": null,
        "dod_claimant_program_description": null,
        "other_than_full_and_open": null,
        "domestic_or_foreign_entity": null,
        "extent_competed": "A",
        "construction_wage_rate": "X",
        "referenced_idv_agency_iden": "1448",
        "solicitation_procedures_description": null,
        "commercial_item_test_program": "N",
        "information_technology_commercial_item_category": null,
        "clinger_cohen_act_planning": "N: NO",
        "information_technology_commercial_item_category_description": null,
        "fair_opportunity_limited": null,
        "idv_type_description": null,
        "type_set_aside": "SBA",
        "program_acronym": null,
        "construction_wage_rate_description": "NOT APPLICABLE",
        "fed_biz_opps": "Y",
        "purchase_card_as_payment_method_description": "NO",
        "naics_description": "DIRECT TITLE INSURANCE CARRIERS",
        "sea_transportation": null,
        "consolidated_contract_description": "NO"
    },
    "funding_agency": {
        "office_agency_name": null,
        "subtier_agency": {
            "code": "1448",
            "name": "U.S. Fish and Wildlife Service",
            "abbreviation": "USFWS"
        },
        "toptier_agency": {
            "code": "014",
            "name": "Department of the Interior",
            "abbreviation": "DOI"
        },
        "id": 239
    },
    "awarding_agency": {
        "office_agency_name": "CONTRACTING&GENERAL SERVICES DIV.",
        "subtier_agency": {
            "code": "1448",
            "name": "U.S. Fish and Wildlife Service",
            "abbreviation": "USFWS"
        },
        "toptier_agency": {
            "code": "014",
            "name": "Department of the Interior",
            "abbreviation": "DOI"
        },
        "id": 239
    },
    "period_of_performance": {
        "start_date": "2000-11-15",
        "end_date": "2001-01-15",
        "last_modified_date": "2001-04-30 00:00:00",
        "potential_end_date": "2001-01-15 00:00:00"
    },
    "recipient": {
        "recipient_hash": null,
        "recipient_name": "TSI TITLE SERVICES INC",
        "recipient_unique_id": "859881898",
        "parent_recipient_hash": null,
        "parent_recipient_name": "PHILLIPS LAW FIRM PA",
        "parent_recipient_unique_id": "859881898",
        "business_categories": [
            "Small Business",
            "Category Business"
        ],
        "location": {
            "location_country_code": "UNITED STATES",
            "country_name": null,
            "state_code": "MS",
            "city_name": "HATTIESBURG",
            "county_name": null,
            "address_line1": "7 WOODSTONE PLZ STE 7",
            "address_line2": null,
            "address_line3": null,
            "congressional_code": "MS04",
            "zip4": null,
            "zip5": null,
            "foreign_postal_code": null,
            "foreign_province": null
        }
    },
    "executive_details": {
        "officers": [
            {
                "name": null,
                "amount": null
            },
            {
                "name": null,
                "amount": null
            },
            {
                "name": null,
                "amount": null
            },
            {
                "name": null,
                "amount": null
            },
            {
                "name": null,
                "amount": null
            }
        ]
    },
    "place_of_performance": {
        "location_country_code": "USA",
        "country_name": "UNITED STATES",
        "county_name": null,
        "city_name": null,
        "state_code": "MS",
        "state_name": "MISSISSIPPI",
        "congressional_code": "00",
        "zip4": null,
        "zip5": null,
        "address_line1": null,
        "address_line2": null,
        "address_line3": null,
        "foreign_province": null,
        "foreign_postal_code": null
    },
    "psc_hierarchy": {
        "toptier_code": {
            "code": "R",
            "description": "SUPPORT SVCS (PROF, ADMIN, MGMT)"
        },
        "subtier_code": {},
        "base_code": {
            "code": "R402",
            "description": "SUPPORT- PROFESSIONAL: REAL ESTATE BROKERAGE"
        },
        "midtier_code": {
            "code": "R4",
            "description": "PROFESSIONAL SERVICES"
        }
    },
    "naics_hierarchy": {
        "toptier_code": {
            "code": "52",
            "description": "Finance and Insurance"
        },
        "base_code": {
            "code": "524127",
            "description": "Direct Title Insurance Carriers"
        },
        "midtier_code": {
            "code": "5241",
            "description": "Insurance Carriers"
        }
    },
    "parent_generated_unique_award_id": "CONT_IDV_401819D010_1448"
}
```

### CODE REVIEW

File: usaspending_api/awards/v2/data_layer/orm.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 from collections import OrderedDict
 from decimal import Decimal
-from django.db.models import Sum
+from django.db.models import Sum, F
```
Comment (BrianZito) [inline usaspending_api/awards/v2/data_layer/orm.py]: I know it's Django, not you, but `F`?

Comment (sethstoudenmier) [reply usaspending_api/awards/v2/data_layer/orm.py]: The long version is in [Django Docs](https://docs.djangoproject.com/en/1.11/ref/models/expressions/#f-expressions).

The short version is that it is a workaround to do aliases with Django when you are just grabbing a column. In this use case it allows the Contract and IDV response to both use the same function to fetch the parent award details even though the query paths would be different: `award_id` for Contract vs `parent_award__award_id` for IDVs.

File: usaspending_api/awards/v2/data_layer/orm.py [hunk 2]
```diff
@@ -256,31 +253,43 @@ def fetch_award_details(filter_q: dict, mapper_fields: OrderedDict) -> dict:
...
-        parent = (
-            ParentAward.objects.filter(generated_unique_award_id=parent_unique_key)
-            .values("generated_unique_award_id")
-            .first()
-        )
-        if parent:
-            return parent["generated_unique_award_id"]
-    return None
+def fetch_contract_parent_award_details(parent_piid: str, parent_fpds_agency: str) -> Optional[OrderedDict]:
+    parent_guai = "CONT_IDV_{}_{}".format(parent_piid or "NONE", parent_fpds_agency or "NONE")
```
Comment (BrianZito) [inline usaspending_api/awards/v2/data_layer/orm.py]: I would advise against us getting used to using `guai` to mean `generated unique award id`. It's not exactly obvious.

Comment (BrianZito) [reply usaspending_api/awards/v2/data_layer/orm.py]: `unique_id` `gen_unique_id` `generated_id`

`GUAI` is more clearly an acronym.

**Comment by tony-sappe** [comment]
Since this will break some award pages, this PR merge should wait until frontend has a complementary PR ready.

**Pull request merged**